### PR TITLE
feat: add mul_q and div_q

### PR DESCRIPTION
**Pull request by wjw12**

### CODE REVIEW

File: sources/uq64x64.move [hunk 1]
```diff
@@ -106,6 +106,28 @@ module uq64x64::uq64x64 {
...
+        UQ64x64{ v }
+    }
+
+    
+    /// Divide a `UQ64x64` by a `UQ64x64`, returning a `UQ64x64`.
+    public fun div_q(a: UQ64x64, b: UQ64x64): UQ64x64 {
+        assert!(b.v != 0, ERR_DIVIDE_BY_ZERO);
+        // vm would direct abort when overflow occured
+        let a_shift = a.v << 32;
+        let b_shift = b.v >> 32;
```
Comment (wjw12) [inline sources/uq64x64.move:153]: need to check b_shift > 0

File: tests/uq64x64_tests.move [hunk 2]
```diff
@@ -146,4 +146,49 @@ module uq64x64::uq64x64_tests {
...
+    
+    #[test]
+    fun test_div_q_fraction() {
+        let a = uq64x64::fraction(5, 4); // 1.25
+        let b = uq64x64::fraction(1, 4); // 0.25
+        let z = uq64x64::div_q(a, b);
+        assert!(uq64x64::to_u128(z) == TWO_POWER_64 * 5, 0);
+        // truncation should happen
+        assert!(uq64x64::decode(z) == 5, 1);
+    }
```
Comment (wjw12) [inline tests/uq64x64_tests.move:212]: need to add div_q overflow case

File: sources/uq64x64.move (hunk 1 above)
Comment (wjw12) [reply sources/uq64x64.move:153]: done

File: tests/uq64x64_tests.move (hunk 2 above)
Comment (wjw12) [reply tests/uq64x64_tests.move:212]: done

File: README.md [hunk 3]
```diff
@@ -18,8 +18,8 @@ Add to `Move.toml`:
 
 ```toml
 [dependencies.UQ64x64]
-git = "https://github.com/pontem-network/UQ64x64.git"
-rev = "v0.3.6"
+git = "https://github.com/ThalaLabs/UQ64x64.git"
+rev = "v0.3.7"
```
Comment (0xbe1) [inline README.md]: we are not releasing as pontem does https://github.com/pontem-network/uq64x64/releases/tag/v0.3.6
so probably just use commit hash here

**Review by 0xbe1 — APPROVED** [review]
lgtm

File: README.md (hunk 3 above)
Comment (wjw12) [reply README.md]: agreed. Maintaining releases seems to be too much overhead

**Pull request merged**